pytest/tests/sanity/rpc_hash.py

```diff
@@ -0,0 +1,199 @@
+    inner_rest.last_ds_final_block = get_hash('last_ds_final_block')
+    inner_rest.block_ordinal = get_int('block_ordinal')
+    inner_rest.prev_height = get_int('prev_height')
+    # TODO: Handle non-None values.
+    inner_rest.epoch_sync_data_hash = header['epoch_sync_data_hash']
+    inner_rest.approvals = [messages.crypto.Signature(approval)
+                            for approval in header['approvals']]
+    inner_rest.latest_protocol_version = get_int('latest_protocol_version')
+    inner_rest_blob = serialize(inner_rest)
+    inner_rest_hash = sha256(inner_rest_blob)
+
+    inner_hash = sha256(inner_lite_hash + inner_rest_hash)
+    block_hash = sha256(inner_hash + prev_hash)
+
+    logger.info('; '.join((name + ': ' + base58.b58encode(h).decode('ascii')
+                          for name, h in (
+                          ('hash_lite', inner_lite_hash),
+                          ('hash_rest', inner_rest_hash),
+                          ('inner_hash', inner_hash),
+                          ('prev_hash', prev_hash),
+                          ('block_hash', block_hash),
+                          ))))
+
+    return base58.b58encode(block_hash).decode('ascii')
+
+
+class HashTestCase(unittest.TestCase):
+    _nodes: typing.Sequence[cluster.BaseNode]
+
+    def __init__(self, *args, **kw) -> None:
+        super().__init__(*args, **kw)
+        self._nodes = ()
+        self.maxDiff = None
+
+    def _test_block_hash(self,
+                         msg_version: int,
+                         protocol_version: typing.Optional[int] = None) -> None:
+        """Starts a cluster, fetches the first block and computes its hash.
+
+        Args:
+            msg_version: Version of the BlockHeaderInnerRest to use when
+                serialising and computing hash.
+            protocol_version: If given, protocol version to use in the cluster
+                (which will be set in genesis); If not given, cluster will be
+                started with the newest supported protocol version.
+        """
+        genesis_overrides = []
+        if protocol_version:
+            genesis_overrides = [['protocol_version', protocol_version]]
+
+        nodes = ()
+        try:
+            nodes = cluster.start_cluster(2, 0, 1, None, genesis_overrides, {})
+            node = nodes[0]
+            block_id = utils.wait_for_blocks(node, target=1)
+            header = node.get_block(block_id.hash)['result']['header']
+        finally:
+            for node in nodes:
+                node.cleanup()
+
+        #logger.info('header = %s', json.dumps(header, indent=True))
+        self.assertEqual((block_id.height, block_id.hash),
+                         (header['height'], header['hash']),
+                         (block_id, header))
+
+        got = compute_block_hash(header, msg_version)
+        self.assertEqual(header['hash'], got)
+
+    def test_block_hash_v1(self):
+        """Starts a cluster, fetches the first block and computes its hash.
```

Comment:
could we test a non-first block too?

Reply:
It actually doesn’t matter much.  In practice the tests takes ~4th block.